src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -60,6 +61,18 @@ public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 cer
             return this;
         }
 
+        /// <summary>
+        /// Sets the certificate associated with the application
+        /// </summary>
+        /// <param name="certificate">The X509 certificate used as credentials to prove the identity of the application to Azure AD.</param>
+        /// <remarks>You should use certificates with a private key size of at least 2048 bytes. Future versions of this library might reject certificates with smaller keys. </remarks>
+        public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 certificate, Dictionary<string, string> claimsToSign)
+        {
+            Config.ClientCredentialCertificate = certificate;
```

Comment:
THere are missing arg null validatins in all these classes. If a developer passes in a null certificate at this stage - throw. Empty claims list - throw.

Reply:
from what i see in the code it looks like all of the argument null catching is done in the ClientAssertionCertificateWrapper class